### PR TITLE
docs: deprecate eslint v8

### PR DESCRIPTION
**Pull request by MikeMcC399**
## Issue

[ESLint v8.x end-of-life](https://eslint.org/version-support/) begins on Oct 5, 2024 (Edit: this is now in the past).

The [eslint-plugin-cypress](https://www.npmjs.com/package/eslint-plugin-cypress) plugin needs to withdraw support for ESLint v8 due to this status change.

## Change

Add deprecation information for ESLint v8 to existing section [README > Deprecations](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/README.md#deprecations) which already deprecates support for ESLint v7.

There are no technical changes to the plugin caused by the deprecation.

### CODE REVIEW

**Comment by MikeMcC399** [comment]
Preview

https://github.com/MikeMcC399/eslint-plugin-cypress/blob/deprecate/eslint-v8/README.md#deprecations

**Comment by MikeMcC399** [comment]
On hold until Oct 5, 2024.

**Comment by MikeMcC399** [comment]
This PR can be reviewed now and can be merged on or after Oct 5, 2024.

This is just so the wording is correct, which states "ESLint `v8` reached end-of-life on Oct 5, 2024" - so that needs to be in the past (not the future) in order to make sense.

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**
